src/chonkie/friends/handshakes/mongodb.py

```
@@ -0,0 +1,177 @@
+        uri: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        hostname: Optional[str] = None,
+        port: Optional[Union[int, str]] = None,
+        db_name: Union[str, Literal["random"]] = "random",
+        collection_name: Union[str, Literal["random"]] = "random",
+        embedding_model: Union[str, BaseEmbeddings] = "minishlab/potion-retrieval-32M",
+        **kwargs: Any,
+    ) -> None:
+        """Initialize MongoDB Handshake with the specified connection parameters.
+
+        Args:
+            client: The MongoDB client to use. If None, will create a new client.
+            uri: The MongoDB connection URI.
+            username: MongoDB username for authentication.
+            password: MongoDB password for authentication.
+            hostname: MongoDB host address.
+            port: MongoDB port number.
+            db_name: The name of the database or "random" for auto-generated name.
+            collection_name: The name of the collection or "random" for auto-generated name.
+            embedding_model: The embedding model identifier or instance.
+            **kwargs: Additional keyword arguments for MongoDB client.
+
+        """
+        super().__init__()
+        self._import_dependencies()
+
+        if client is not None:
+            self.client = client
+        else:
+            # use uri
+            if uri is None:
+                # construct the uri
+                if (
+                    username is not None
+                    and password is not None
+                    and hostname is not None
+                ):
+                    uri = f"mongodb://{username}:{password}@{hostname}"
+                # use localhost
+                else:
+                    port = str(port) if port is not None else "27017"
+                    uri = f"mongodb://localhost:{port}"
+                    # clear port
+                    port = None
```

Hey @chonk-lain!

Could you throw an error to the user if they provide a `hostname` but no `username` & `password`? I feel like a silent change to `localhost` might result in unknown errors down the line. 

And if switching to `localhost` adding `warning` would be a good idea to let the user know that we're trying to check on `localhost` instead. 

Thanks 😊

---

reconstructed the logic and added a comment in case we are manually falling back to localhost